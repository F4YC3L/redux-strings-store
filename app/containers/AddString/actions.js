/*
 * Home Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  CHANGE_NEW_STRING,
  ADD_STRING_SUCCESS,
  ADD_STRING_ERROR,
  ADD_STRING,
} from './constants';

/**
 * Changes the input field of the form
 *
 * @param  {string} newString The new text of the input field
 *
 * @return {object} An action object with a type of CHANGE_NEW_STRING
 */
export function changeNewString(newString) {
  return {
    type: CHANGE_NEW_STRING,
    newString,
  };
}

/**
 * add string, this action starts the request saga
 *
 * @return {object} An action object with a type of ADD_STRING
 */
export function addString() {
  return {
    type: ADD_STRING,
  };
}

/**
 * add string, this action starts the request saga
 *
 * @return {object} An action object with a type of ADD_STRING_SUCCESS
 */
export function stringAdded() {
  return {
    type: ADD_STRING_SUCCESS,
  };
}

/**
 * Dispatched when loading the data fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of ADD_STRING_ERROR passing the error
 */
export function stringAddingError(error) {
  return {
    type: ADD_STRING_ERROR,
    error,
  };
}
