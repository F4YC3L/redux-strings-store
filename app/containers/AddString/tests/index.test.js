/**
 * Test the HomePage
 */

import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { AddString, mapDispatchToProps } from '../index';
import { changeNewString, addString } from '../actions';
import configureStore from '../../../configureStore';

describe('<AddString />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <AddString loading={false} error={false} newString="" />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('onChangeNewString', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onChangeNewString).toBeDefined();
      });

      it('should dispatch changeNewString when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const newString = 'JavaScript';
        result.onChangeNewString({ target: { value: newString } });
        expect(dispatch).toHaveBeenCalledWith(changeNewString(newString));
      });
    });

    describe('onSubmitForm', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onSubmitForm).toBeDefined();
      });

      it('should dispatch addString when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onSubmitForm();
        expect(dispatch).toHaveBeenCalledWith(addString());
      });

      it('should preventDefault if called with event', () => {
        const preventDefault = jest.fn();
        const result = mapDispatchToProps(() => {});
        const evt = { preventDefault };
        result.onSubmitForm(evt);
        expect(preventDefault).toHaveBeenCalledWith();
      });
    });
  });
});
