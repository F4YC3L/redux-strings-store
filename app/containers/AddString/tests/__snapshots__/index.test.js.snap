// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddString /> should render and match the snapshot 1`] = `
<article>
  <section
    class="Section-sc-18n5ohj-0 dPYDU"
  >
    <form
      class="Form-sc-148ipml-0 hVyDgC"
    >
      <label
        for="newString"
      >
        <input
          class="Input-i9rmlz-0 caCtWU"
          id="newString"
          placeholder="Enter something"
          type="text"
          value=""
        />
      </label>
      <button
        class="button-dkh3p2-0 iEwNap"
        disabled=""
      >
        Add
      </button>
    </form>
  </section>
  <section
    class="Section-sc-18n5ohj-0 dPYDU"
  />
</article>
`;
